### DISCUSSION TITLE
'method' object is not iterable

### DISCUSSION THREAD

**Discussion by pratyushdhingra10** [pyg-team/pytorch_geometric#5889] [General]
I am getting below error while using pytorch_geometric to train the PPI database.

Traceback (most recent call last):
  File "PPI_GAT_GD.py", line 20, in <module>
    train_dataset = PPI(path, split='train')
  File "/lib/python3.8/site-packages/torch_geometric/datasets/ppi.py", line 67, in __init__
    super().__init__(root, transform, pre_transform, pre_filter)
  File "/lib/python3.8/site-packages/torch_geometric/data/in_memory_dataset.py", line 50, in __init__
    super().__init__(root, transform, pre_transform, pre_filter)
  File "/lib/python3.8/site-packages/torch_geometric/data/dataset.py", line 87, in __init__
    self._process()
  File "//lib/python3.8/site-packages/torch_geometric/data/dataset.py", line 170, in _process
    self.process()
  File "/lib/python3.8/site-packages/torch_geometric/datasets/ppi.py", line 116, in process
    edge_index = torch.tensor(list(G_s.edges)).t().contiguous()

TypeError: 'method' object is not iterable

Version Used: 
torch-cluster             1.6.0   
torch-geometric       2.0.4 
torch-scatter             2.0.9   
torch                         1.11.0

**Comment by anthonysirico** [comment]
Can you put in that section of code?

**Comment by pratyushdhingra10** [comment]
Hi, It is just crashing at the step of loading the database


import os.path as osp
import os
import copy
import re
import torch
import torch.nn.functional as F
import utils_prune
import utils
import numpy as np
from torch import linalg as LA

from torch_geometric.datasets import PPI
from torch_geometric.loader import DataLoader
from torch_geometric.nn import GATConv

path = osp.join(osp.dirname(osp.realpath(__file__)), '..', 'data', 'PPI')
train_dataset = PPI(path, split='train')
val_dataset = PPI(path, split='val')
test_dataset = PPI(path, split='test')

**Reply by anthonysirico** [reply]
Strange, its working for me.
```
from torch_geometric.datasets import PPI

dataset = PPI(root='PPI', split='train')


Downloading https://data.dgl.ai/dataset/ppi.zip
Extracting PPI\ppi.zip
Processing...
Done!
```
```
import torch_cluster
import torch_scatter
import torch_geometric
print(torch_cluster.__version__)
print(torch_scatter.__version__)
print(torch.__version__)
print(torch_geometric.__version__)

1.6.0
2.0.9
1.12.1
2.1.0
```

**Reply by pratyushdhingra10** [reply]
That is strange. Any guesses what could be the issue ?

**Reply by anthonysirico** [reply]
That would be a developer question, sorry, and they are on euro time.

**Reply by rusty1s** [reply]
What's the `networkx` version you are using? Can you try to upgrade to the latest one? It works for me on `networkx==2.8.*`.

**Reply by pratyushdhingra10** [reply]
Hi, Changing the networkx version worked. Thanks :)

**Comment by BraveDrXuTF** [comment]
> What's the `networkx` version you are using? Can you try to upgrade to the latest one? It works for me on `networkx==2.8.*`.

Hi, I got the same problem but with networkx==2.8.7, and I just found this discussion. My error code snippet:
```python
# see https://github.com/sungyongs/dpgn/blob/master/utils.py
def decompose_graph(graph):
    # graph: torch_geometric.data.data.Data
    # TODO: make it more robust
    x, edge_index, edge_attr, global_attr = None, None, None, None
    for key in graph.keys:
        if key=="x":
            x = graph.x
        elif key=="edge_index":
            edge_index = graph.edge_index
        elif key=="edge_attr":
            edge_attr = graph.edge_attr
        elif key=="global_attr":
            global_attr = graph.global_attr
        else:
            pass
    return (x, edge_index, edge_attr, global_attr)
```
where the line `for key in graph.keys` got this error, any idea? I would be grateful if you provide any feedback.

**Comment by BraveDrXuTF** [comment]
> What's the `networkx` version you are using? Can you try to upgrade to the latest one? It works for me on `networkx==2.8.*`.

My problem has been solved, just add a（）.

**Reply by Gina123e** [reply]
shere the () ia added to

**Reply by Gina123e** [reply]
graph.key.() or graph.().key()
